core/blockarchiver/client.go

```diff
@@ -0,0 +1,229 @@
+package blockarchiver
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	bundlesdk "github.com/bnb-chain/greenfield-bundle-sdk/bundle"
+)
+
+// Client is a client to interact with the block archiver service
+type Client struct {
+	hc                *http.Client
+	blockArchiverHost string
+	spHost            string
+	bucketName        string
+}
+
+func New(blockAchieverHost, spHost, bucketName string) (*Client, error) {
```

Comment:
As we discussed, the client will request from Bundle service directly instead of `blockAchieverHost`?